player/lua/console.lua

```diff
@@ -1096,38 +1096,18 @@ local function get_clipboard(clip)
             return res.stdout
         end
     elseif platform == 'wayland' then
-        local res = utils.subprocess({
-            args = { 'wl-paste', clip and '-n' or  '-np' },
-            playback_only = false,
-        })
-        if not res.error then
-            return res.stdout
+        if clip and mp.get_property('current-vo') and mp.get_property_native('video-osd') then
```

Comment:
Is those checks required? Wouldn't it be better to check `mp.get_property('clipboard/text', nil) ~= nil`  and fallback to wl-paste?

Reply:
I see. And it cannot be detected on clipboard side? Well, it seems bit tricky for script users to know this detail.

Also would `focused` property work in this case? It is implemented by wayland.

Reply:
The clipboard property still works with a VO and just --video-osd=no.

Reply:
Yes, please. It should be noted in docs.

Reply:
There is no need to check `current-vo` if `focused` is true, it can't be true in the terminal.